lang/csharp/src/apache/test/Schema/SchemaTests.cs

```diff
@@ -69,8 +69,12 @@ public class SchemaTests
             typeof(SchemaParseException), Description = "No fields")]
         [TestCase("{\"type\":\"record\",\"name\":\"LongList\", \"fields\": \"hi\"}",
             typeof(SchemaParseException), Description = "Fields not an array")]
-        [TestCase("[{\"type\": \"record\",\"name\": \"Test\",\"namespace\":\"ns1\",\"fields\": [{\"name\": \"f\",\"type\": \"long\"}]}," + 
+        [TestCase("[{\"type\": \"record\",\"name\": \"Test\",\"namespace\":\"ns1\",\"fields\": [{\"name\": \"f\",\"type\": \"long\"}]}," +
                    "{\"type\": \"record\",\"name\": \"Test\",\"namespace\":\"ns2\",\"fields\": [{\"name\": \"f\",\"type\": \"long\"}]}]")]
+
+        // Doc
+        [TestCase("{\"type\": \"record\",\"name\": \"Test\",\"doc\": \"Test Doc\",\"fields\": [{\"name\": \"f\",\"type\": \"long\"}]}")]
+
```

Comment:
There are a few doc tests in this file already like `TestRecordDoc` and `TestEnumDoc`. These check that the field is parsed in without error, which is all that this case does as well. None of them actually check that the doc string is written when serialising though.

Changing these existing tests to do something like write the schema out and parse it again will give you test cases for the whole cycle which I think is quite important to this PR, something like:

```
        public void TestRecordDoc(string s, string expectedDoc)
        {
            var parsed = Schema.Parse(s) as RecordSchema;

            Assert.IsNotNull(parsed);
            Assert.AreEqual(expectedDoc, parsed.Documentation);

            var roundTrip = Schema.Parse(parsed.ToString()) as RecordSchema;

            Assert.IsNotNull(roundTrip);
            Assert.AreEqual(expectedDoc, roundTrip.Documentation);
        }
```

I'm not clear on how empty strings should be treated though.

Reply:
Excellent point.  I updated the test to use your round trip logic.  As far as empty strings, my original code would have eliminated them from the output, but since the test was written expecting them, I changed it to preserve the doc attribute when specified; empty or not.